dlib/dnn/loss.h

```diff
@@ -990,21 +1035,34 @@ namespace dlib
 
         size_t find_best_detection_window (
             rectangle rect,
-            const std::string& label
+            const std::string& label,
+            bool scale_invariant
         ) const
         {
-            rect = move_rect(set_rect_area(rect, 400*400), point(0,0));
+            if (scale_invariant)
+            {
+                rect = move_rect(set_rect_area(rect, 400*400), point(0,0));
+            }
+            else
+            {
+                rect = rectangle(rect.width(), rect.height());
+            }
 
-            // Figure out which detection window in options.detector_windows has the most
-            // similar aspect ratio to rect.
+            // Figure out which detection window in options.detector_windows is most similar to rect
+            // (in terms of aspect ratio, if scale_invariant == true).
             size_t best_i = 0;
             double best_ratio_diff = -std::numeric_limits<double>::infinity();
             for (size_t i = 0; i < options.detector_windows.size(); ++i)
             {
                 if (options.detector_windows[i].label != label)
                     continue;
-                rectangle det_window = centered_rect(point(0,0), options.detector_windows[i].width, options.detector_windows[i].height);
-                det_window = move_rect(set_rect_area(det_window, 400*400), point(0,0));
+
+                rectangle det_window(options.detector_windows[i].width, options.detector_windows[i].height);
+
```

Comment:
This code is a little confusing.  Put it in an if/else rather than a default assignment and an if that overrides.

Reply:
Fixed.